src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -576,6 +576,14 @@ std::shared_ptr<IExecutableNetworkInternal> IInferRequestInternal::getPointerToE
     return _exeNetwork;
 }
 
+void IInferRequestInternal::setPointerToSo(const std::shared_ptr<void>& so) {
+    _so = so;
+}
+
+std::shared_ptr<void> IInferRequestInternal::getPointerToSo() const {
+    return _so;
```

Comment:
`IInferRequestInternal` has a reference to a `IExecutableNetworkInternal` which created this infer request (see https://github.com/openvinotoolkit/openvino/blob/master/src/inference/dev_api/cpp_interfaces/interface/ie_iinfer_request_internal.hpp#L352). So, can we remove manual `IInferRequestInternal::setPointerToSo` and `IExecutableNetworkInternal::GetPointerToSo` and infer request can take `_so` from the exec network internal field directly?

Reply:
yes for normal cases, but in this case for virtual plugin like HETERO, this hetero infer request is holding reference of hetero plugin, yet when user get tensor from hetero, hetero will pass the hw memory, etc..gpu blob directly to user, we will need to hold gpu related resources as well in case that this gpu tensor is somehow removed at the very end(after compiled model and ie core)